webprotege-gwt-ui-client/src/main/java/edu/stanford/bmir/protege/web/client/hierarchy/EntityHierarchyDropHandler.java

```diff
@@ -127,8 +127,12 @@ public void handleDrop(@Nonnull Path<EntityNode> nodePath,
                                         });
                                 return;
                             }
-                            if (moveResult.isDestinationRetiredClass()) {
-                                messageBox.showMessage(messages.classHierarchy_cannotMoveReleasedClassToRetiredParent());
+                            if (moveResult.isDestinationRetiredClass() || (moveResult.getReleasedChildrenValidationMessage() != null && !moveResult.getReleasedChildrenValidationMessage().isEmpty())) {
+                                String errorMessage = messages.classHierarchy_cannotMoveReleasedClassToRetiredParent();
+                                if(moveResult.getReleasedChildrenValidationMessage() != null && !moveResult.getReleasedChildrenValidationMessage().isEmpty()) {
+                                    errorMessage = errorMessage + ". " + moveResult.getReleasedChildrenValidationMessage();
```

Comment:
I don't think we need this period. The error message usually ends with a punctuation mark (from what I've seen an exclamation mark), which does not need to be followed by a period. You could use a new line character (if is to be sent to the API) or a <BR> if it is for HTML rendering, or simply just a space, multiple spaces or a TAB.